### PR TITLE
Missing title in engine/tutorials/index.md - add framework for CI checking

### PR DESCRIPTION
**Pull request by aduermael**
I added a missing title in `engine/tutorials/index.md`

<s>Also created `hook` folder and `pre_build` script for Docker Cloud CI. 
We should turn on autotest for pull requests in build configurations. (cc @johndmulhausen )</s>

Also added a `tests` folder that contains a Dockerfile and go code defining tests for that repo. 
It comes with a **Jenkinsfile** to automate them.

That PR includes one first test to validate content. It simply checks if there's a title in all markdown frontmatters. (thanks @gdevillele for your frontparser package)

We should add more tests as we fix issues in content.

### CODE REVIEW

**Comment by johndmulhausen** [comment]
Deploy preview ready!

Built with commit d513940b400b533fb8a24ddcd8460c560cf3dcf6

https://deploy-preview-763--docker-docs.netlify.com

**Comment by aduermael** [comment]
Unfortunately we can't run tests for PRs using Docker Cloud, public repos are not authorized (yet). 
But it's fine, I'll setup Jenkins while waiting for it.
@johndmulhausen please keep that PR open while we test integration.

**Comment by aduermael** [comment]
ok to test

**Comment by aduermael** [comment]
test this please

**Comment by aduermael** [comment]
ok to test

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+wrappedNode(label: 'linux && x86_64') {
+  deleteDir()
+  stage "checkout"
+  checkout scm
+  stage "test"
+  sh "ls -al"
+  bash hooks/pre_build
```
Comment (mikedougherty) [inline Jenkinsfile]: `sh "hooks/pre_build"`

**Comment by aduermael** [comment]
@johndmulhausen that PR is ready to be merged. 
But after you merge it, we should update 2 things in the settings: 
- Protect master branch (Disables force-pushes to this branch and prevents it from being deleted.)
- Require status checks to pass before merging

**Comment by johndmulhausen** [comment]
Thanks, @aduermael and thanks for demoing this to me. This is a great step for the docs, getting a doc CI framework. :)

**Pull request merged**